sdk/python/tests/compiler/testdata/resourceop_basic.py

```diff
@@ -57,7 +57,9 @@ def resourceop_basic():
     op = dsl.ResourceOp(
         name='test-step',
         k8s_resource=json.loads(_CONTAINER_MANIFEST),
-        action='create'
+        action='create',
+        success_condition='status.succeeded > 0',
```

Comment:
I just found out you changed this test case, we shouldn't add success_condition and failure_condition because they suppose to be using the default param with `''`.

Reply:
@Tomcli -- they failed Tekton validation without it:

```invalid input params: missing values for these params which have no default values: [success-condition failure-condition]```

Reply:
@Tomcli -- why are `''` empty string non-default default parameters required here instead of actual `success_condition` and `failure_condition` ?

Reply:
We don't have to set the actual success_condition and failure_condition because they should be optional not required. I think the actual issue is at below where the empty string default param are missing.

Reply:
Thanks @ckadner, can you revert the resourceop and volumeop test case without the `success_condition` and `failure_condition`. This way we can catch those edge cases if we see errors again.

Reply:
I removed the `success_condition` and `failure_condition` again